### PR TITLE
Fix #14833: split_part follow pg

### LINKED ISSUE #14833: `split_part` does not return `null` when first input is `null`
### What happens?

[`split_part`](https://duckdb.org/docs/sql/functions/char.html#split_partstring-separator-index) takes three arguments. The first is the string to split. The documentation states that index outside of range behavior for this function is designed to mimic Postgres behavior (and it does). However, Postgres returns `null` when the first input is `null` but DuckDB returns the empty string. This can cause hard-to-diagnose downstream effects, especially when the result of a `split_part` is used as a join key (empty strings match with other empty strings, but `null`s do not match with other `null`s).

### To Reproduce

Simple example:
```sql
.nullvalue NULL
create table foo (a varchar);
insert into foo values ('hello$there'), ('apples'), (''), (null);
select * from foo;
```
```
┌─────────────┐
│      a      │
│   varchar   │
├─────────────┤
│ hello$there │
│ apples      │
│             │
│ NULL        │
└─────────────┘
```
```sql
select count(*) from foo where a is not null;
```
```
┌──────────────┐
│ count_star() │
│    int64     │
├──────────────┤
│            3 │
└──────────────┘
```
```sql
select split_part(a, '$', -1) as s from foo;
```
```
┌─────────┐
│    s    │
│ varchar │
├─────────┤
│ there   │
│ apples  │
│         │
│         │
└─────────┘
```
```sql
select count(*) from foo where split_part(a, '$', -1) is not null;
```
```
┌──────────────┐
│ count_star() │
│    int64     │
├──────────────┤
│            4 │
└──────────────┘
```

Note that since the result of the last query was 4, the function value on `null` input was the empty string, not `null`. This differs from Postgres behavior as shown below:

```sql
postgres: create table foo (a varchar);
CREATE TABLE
postgres: insert into foo values ('hello$there'), ('apples'), (''), (null);
INSERT 0 4
postgres: select count(*) from foo where a is not null;
 count
-------
     3
(1 row)

postgres: select split_part(a, '$', -1) as s from foo;
   s
--------
 there
 apples


(4 rows)

postgres: select count(*) from foo where split_part(a, '$', -1) is not null;
 count
-------
     3
(1 row)
```

### OS:

Manjaro Linus x86_64

### DuckDB Version:

v1.1.1 af39bd0dcf

### DuckDB Client:

CLI

### Hardware:

_No response_

### Full Name:

Doeke Buursma

### Affiliation:

UDisc, LLC

### What is the latest build you tested with? If possible, we recommend testing with the latest nightly build.

I have tested with a stable release

### Did you include all relevant data sets for reproducing the issue?

Not applicable - the reproduction does not require a data set

### Did you include all code required to reproduce the issue?

- [X] Yes, I have

### Did you include all relevant configuration (e.g., CPU architecture, Python version, Linux distribution) to reproduce the issue?

- [X] Yes, I have

### PR DESCRIPTION
**Pull request by flashmouse**
This pr try to fix #14833

### CODE REVIEW

File: src/catalog/default/default_functions.cpp [hunk 1]
```diff
@@ -102,12 +102,12 @@ static const DefaultMacro internal_macros[] = {
...
 	// Test default parameters
 	{DEFAULT_SCHEMA, "array_to_string_comma_default", {"arr", nullptr}, {{"sep", "','"}, {nullptr, nullptr}}, "list_aggr(arr::varchar[], 'string_agg', sep)"},
-	
+
 	{DEFAULT_SCHEMA, "generate_subscripts", {"arr", "dim", nullptr}, {{nullptr, nullptr}}, "unnest(generate_series(1, array_length(arr, dim)))"},
 	{DEFAULT_SCHEMA, "fdiv", {"x", "y", nullptr}, {{nullptr, nullptr}}, "floor(x/y)"},
 	{DEFAULT_SCHEMA, "fmod", {"x", "y", nullptr}, {{nullptr, nullptr}}, "(x-y*floor(x/y))"},
 	{DEFAULT_SCHEMA, "count_if", {"l", nullptr}, {{nullptr, nullptr}}, "sum(if(l, 1, 0))"},
-	{DEFAULT_SCHEMA, "split_part", {"string", "delimiter", "position", nullptr}, {{nullptr, nullptr}}, "coalesce(string_split(string, delimiter)[position],'')"},
+	{DEFAULT_SCHEMA, "split_part", {"string", "delimiter", "position", nullptr}, {{nullptr, nullptr}}, "if(string IS NOT NULL AND delimiter IS NOT NULL, coalesce(string_split(string, delimiter)[position],''), NULL)"},
```
Comment (taniabogatsch) [inline src/catalog/default/default_functions.cpp:110]: Postgres also returns `NULL`, if the position is `NULL`. Let's also align that: 
`if(string IS NOT NULL AND delimiter IS NOT NULL AND position IS NOT NULL, ...`


```sql
postgres=# select split_part('a,b,c','',null) is null;
 ?column? 
----------
 t
(1 row)
```

DuckDB
```sql
D select split_part('a,b,c','',null) is null;
┌─────────────────────────────────────────┐
│ (split_part('a,b,c', '', NULL) IS NULL) │
│                 boolean                 │
├─────────────────────────────────────────┤
│ false                                   │
└─────────────────────────────────────────┘
```

File: test/sql/function/string/test_split_part.test [hunk 2]
```diff
@@ -73,17 +73,17 @@ select split_part('',',',1)
...
 select split_part('a,b,c',NULL,1)
 ----
-a,b,c
+NULL
 
 query T
 select split_part(NULL,',',1)
 ----
-(empty)
+NULL
```
Comment (taniabogatsch) [inline test/sql/function/string/test_split_part.test:86]: Let's add one more test for the 'position is `NULL`' case.

**Review by taniabogatsch — CHANGES_REQUESTED** [review]
Hi @flashmouse, thanks for the PR! Looks good; I've left two comments.

**Review by taniabogatsch — APPROVED** [review]

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**